corehq/apps/domain/views/settings.py

```diff
@@ -271,6 +273,71 @@ def post(self, request, *args, **kwargs):
         return self.get(request, *args, **kwargs)
 
 
+class EditIPAccessConfigView(BaseProjectSettingsView):
+    template_name = 'domain/admin/ip_access_config.html'
+    urlname = 'ip_access_config'
+    page_title = gettext_lazy("IP Access")
+
+    @method_decorator(always_allow_project_access)
+    @method_decorator(login_and_domain_required)
+    @use_bootstrap5
+    def dispatch(self, *args, **kwargs):
+        if IP_ACCESS_CONTROLS.enabled(self.domain):
+            return super().dispatch(*args, **kwargs)
+        else:
+            raise Http404()
```

Comment:
Nit: I'd replace all this with a class-level decorator like:
```python
@method_decorator([
    always_allow_project_access,
    login_and_domain_required,
    use_bootstrap5,
    toggles.IP_ACCESS_CONTROLS.required_decorator(),
], name='dispatch')
class EditIPAccessConfigView(BaseProjectSettingsView):
```
No need to override `dispatch`

Reply:
Also what's the `always_allow_project_access` decorator doing here?  I think that lets this page be accessed when the subscription is paused, but I see it's pretty widely used.

And I think this needs `domain_admin_required` instead of `login_and_domain_required`